litex/tools/litex_soc_gen.py

```diff
@@ -62,56 +63,66 @@ def build(self, fragment, build_dir, build_name, **kwargs):
 # LiteX SoC Generator ------------------------------------------------------------------------------
 
 class LiteXSoCGenerator(SoCMini):
-    def __init__(self, name="litex_soc", sys_clk_freq=int(50e6), **kwargs):
+    def __init__(self, name="litex_soc", sys_clk_freq=int(50e6), n_mi=3, n_si=0, **kwargs):
+        assert (n_mi + n_si) < 5
+        self.kwargs = kwargs
+
         # Platform ---------------------------------------------------------------------------------
-        platform = Platform(device="", io=get_common_ios())
-        platform.name = name
-        platform.add_extension(get_uart_ios())
+        self.platform = Platform(device="", io=get_common_ios())
+        self.platform.name = name
+        self.platform.add_extension(get_uart_ios())
 
         # CRG --------------------------------------------------------------------------------------
         self.submodules.crg = CRG(
-            clk = platform.request("clk"),
-            rst = platform.request("rst"),
+            clk = self.platform.request("clk"),
+            rst = self.platform.request("rst"),
         )
 
         # SoC --------------------------------------------------------------------------------------
-        if kwargs["uart_name"] == "serial":
-            kwargs["uart_name"] = "uart"
-        SoCCore.__init__(self, platform, clk_freq=sys_clk_freq, ident=f"LiteX standalone SoC - {name}", **kwargs)
+        if self.kwargs["uart_name"] == "serial":
+            self.kwargs["uart_name"] = "uart"
+        SoCCore.__init__(self, self.platform, clk_freq=sys_clk_freq, ident=f"LiteX standalone SoC - {name}", **kwargs)
 
-        # MMAP Slave Interface ---------------------------------------------------------------------
-        s_bus = {
-            "wishbone" : wishbone.Interface(),
-            "axi-lite" : axi.AXILiteInterface(),
+        for n in range(n_mi):
+            origins=[0xa000_0000, 0xb000_0000, 0xc000_0000, 0xd000_0000]
```

Comment:
If we expose the number of interface, we should also probably allow configuration of base addresses. We can wait to address the other points before looking at it.